public/js/data.js

```diff
@@ -767,27 +767,32 @@ createOpenLocalPrompt = function() {
 // Prompt to create subcircuit, shows list of circuits which dont depend on the current circuit
 createSubCircuitPrompt = function(scope = globalScope) {
     $('#insertSubcircuitDialog').empty();
+    
     var flag = true;
+
     for (id in scopeList) {
         if (!scopeList[id].checkDependency(scope.id)) {
             flag = false;
             $('#insertSubcircuitDialog').append('<label class="option"><input type="radio" name="subCircuitId" value="' + id + '" />' + scopeList[id].name + '</label>');
         }
     }
+
     if (flag) $('#insertSubcircuitDialog').append('<p>Looks like there are no other circuits which doesn\'t have this circuit as a dependency. Create a new one!</p>')
+    
     $('#insertSubcircuitDialog').dialog({
         width: "auto",
         buttons: [{
             text: "Insert SubCircuit",
             click: function() {
                 if (!$("input[name=subCircuitId]:checked").val()) return;
-                simulationArea.lastSelected = new SubCircuit(undefined, undefined, globalScope, $("input[name=subCircuitId]:checked").val());
+                let subcircuitId = parseInt($("input[name=subCircuitId]:checked").val(), 10);
+                const savedData = JSON.parse(generateSaveData("Untitled"));
+                const subcircuitData = savedData.scopes.filter((circuitData) => circuitData.id === subcircuitId)
+                simulationArea.lastSelected = new SubCircuit(undefined, undefined, globalScope, subcircuitId, subcircuitData);
```

Comment:
I am not sure what is happening here. However in the SubCircuit module, if the subcircuitData is empty, it instantiates the object. It is meant to be empty when instantiated.

Reply:
I realized that when you insert a subcircuit even if it's not empty, you always get an error that the subcircuit that you are trying to insert is empty. When I tracked the steps a little bit I found that the subcircuit data is 'undefined'. This is why I tried to pass the subcircuit data here.

Reply:
```
 if (subcircuitScope.Input.length == 0 && subcircuitScope.Output.length == 0) {
        showError("SubCircuit : " + subcircuitScope.name + " is an empty circuit");
        this.delete();
        this.scope.backups = [];
        return;
    }
```
I think this condition can be removed but it is there for now. 
This is where it says it is an empty circuit. At least one input or one output should be there. Otherwise that circuit isn't doing anything wrt to the parent circuit. 
Please confirm that this is the error you were getting. Otherwise it is a serious issue that needs to be resolved asap.

Reply:
Thank you for the fast follow up! Yes, this is exactly the condition that is causing the "empty circuit" error that will be displayed when you insert a subcircuit even if has some connections